packages/hooks/use-popper/index.ts

```diff
@@ -0,0 +1,134 @@
+import { computed, onBeforeUnmount, ref, shallowRef, unref, watch } from 'vue'
+import { createPopper } from '@popperjs/core'
+
+import type { Ref } from 'vue'
+import type {
+  Instance,
+  Modifier,
+  Options,
+  State,
+  VirtualElement,
+} from '@popperjs/core'
+
+type ElementType = HTMLElement | undefined
+type ReferenceElement = ElementType | VirtualElement
+
+export const usePopper = (
+  referenceElementRef: Ref<ReferenceElement>,
+  popperElementRef: Ref<ElementType>,
+  opts: Ref<Options> | Options = {} as Options
```

Comment:
Nah, this needs to have a default value for the options so it can work when none passed. It should be `MaybeRef<Options>` here but I think I might want to skip depending on the types of 3rd party libs.